### PR TITLE
Test Importer

### LINKED ISSUE #487: installer add test for successful import
- [x] move start of  standard import process to the end (after adding sample mgmt data)
- [x] run sample imports and check results (fail on error)
- [x] ideally assure that sample data is there, remove again after test import if it did not exist

running manual import with command (as user fworch):
`cd ~/importer; ./fworch-importer-single.pl mgm_id=1`

### PR DESCRIPTION
**Pull request by alf-cactus**
closes #487

### CODE REVIEW

**Review by tpurschke — CHANGES_REQUESTED** [review]
checking for existence of sample data is not enough.
we need to check if the firewall manager and gw actually exist in the database

**Comment by alf-cactus** [comment]
A new testuser is created. This user imports fortigate demos and deletes everything after itself

File: roles/test/tasks/test-importer.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    name: sample-data
+  vars:
+    sample_role_purpose: test
+    sample_fortigate_name: fortigate_test
+    sample_config_user: fworchtest
+    sample_config_user_home: "/home/{{ sample_config_user }}"
+    import_sample_server: localhost
+
+- name: make test import
+  command: "./fworch-importer-single.pl mgm_id=1"
```
Comment (tpurschke) [inline roles/test/tasks/test-importer.yml:13]: should probably be something like mgm_name={{ sample_fortigate_name }} to import the correct system

File: inventory/group_vars/sampleserver.yml [hunk 2]
```diff
@@ -2,9 +2,11 @@
 sample_config_user: fworchsample
 sample_config_user_home: "/home/{{ sample_config_user }}"
 import_sample_server: localhost
-sample_config_hostname: isosrv
-importer_hostname: localhost
+sample_config_hostname: isosrv #todo: delete? not used anywhere
+importer_hostname: localhost 
 sample_data_rate: medium
+sample_role_purpose: demo
+sample_fortigate_name: fortigate_demo
```
Comment (tpurschke) [inline inventory/group_vars/sampleserver.yml:9]: isn't this the same name we already use for the demo system?

File: roles/test/tasks/test-importer.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    state: absent
+    remove: yes
+  become: yes
+
+- name: delete test fortinet gateway
+  postgresql_query:
+    db: "{{ fworch_db_name }}"
+    query: >
+      DO $do$ BEGIN 
+      DELETE FROM device WHERE dev_name='fortigate_test';
```
Comment (tpurschke) [inline roles/test/tasks/test-importer.yml:31]: should we use a variable for the name?

File: roles/test/tasks/test-importer.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  postgresql_query:
+    db: "{{ fworch_db_name }}"
+    query: >
+      DO $do$ BEGIN 
+      DELETE FROM management WHERE mgm_name='fortigate_test'; 
+      END $do$ 
+  become: yes
+  become_user: postgres
+
+- name: make test import
```
Comment (tpurschke) [inline roles/test/tasks/test-importer.yml:46]: what is the purpose of this import?

File: site.yml [hunk 5]
```diff
@@ -99,13 +92,8 @@
...
 
 - hosts: importers
-  tasks:
-    - name: restart importer service to make sure it works correctly with sample data
-      ansible.builtin.systemd:
-        name: "{{ product_name }}-importer"
-        state: restarted
-      become: yes
-      when: "not installation_mode == 'uninstall'"
+  roles:
```
Comment (tpurschke) [inline site.yml:95]: do we install the role importers twice? not necessary imho

**Review by tpurschke — CHANGES_REQUESTED** [review]
see inline comments - let's discuss this after your vacation.

File: roles/test/tasks/test-importer.yml (hunk 1 above)
Comment (alf-cactus) [reply roles/test/tasks/test-importer.yml:13]: introduced a global variable containing the name

File: inventory/group_vars/sampleserver.yml (hunk 2 above)
Comment (alf-cactus) [reply inventory/group_vars/sampleserver.yml:9]: This is the default for the original purpose of the role

File: roles/test/tasks/test-importer.yml (hunk 3 above)
Comment (alf-cactus) [reply roles/test/tasks/test-importer.yml:31]: introduced a global variable containing the name

File: roles/test/tasks/test-importer.yml (hunk 4 above)
Comment (alf-cactus) [reply roles/test/tasks/test-importer.yml]: nothing, I removed it

File: site.yml (hunk 5 above)
Comment (alf-cactus) [reply site.yml:95]: Only once, I moved the role execution

File: ansible.cfg [hunk 6]
```diff
@@ -4,6 +4,7 @@ inventory = inventory
 #timeout = 60
 force_handlers = True
 stdout_callback = yaml
+allow_world_readable_tmpfiles = True
```
Comment (tpurschke) [inline ansible.cfg:7]: what do we need this for?

**Review by tpurschke — COMMENTED** [review]
test install successful - just 
- the question regarding necessity of ansible setting
- an idea about signalling the success of the test import more clearly (maybe using debugging and printing out the result of the import?)

File: ansible.cfg (hunk 6 above)
Comment (alf-cactus) [reply ansible.cfg]: Brought back pipelineing in ansible.cfg, allow_world_readable_tmpfiles not needed anymore

**Review by tpurschke — APPROVED** [review]
ok - let's do it :-)

**Pull request merged**